packages/dd-trace/test/ritm.spec.js

```diff
@@ -29,12 +32,49 @@ describe('Ritm', () => {
 
     moduleLoadStartChannel.subscribe(startListener)
     moduleLoadEndChannel.subscribe(endListener)
-    Hook('module-a')
-    Hook('module-b')
+    const aHook = Hook('module-a')
+    const bHook = Hook('module-b')
     const { a } = require('./ritm-tests/module-a')
 
     assert.equal(startListener.callCount, 2)
     assert.equal(endListener.callCount, 2)
     assert.equal(a(), 'Called by AJ')
+
+    aHook.unhook()
+    bHook.unhook()
+  })
+
+  it('should fall back to monkey patched module', () => {
+    Module.prototype.require = new Proxy(Module.prototype.require, {
+      apply (target, thisArg, argArray) {
+        if (argArray[0] === '@azure/functions-core') {
+          return {
+            version: '1.0.0',
+            registerHook: () => { }
+          }
+        } else {
+          return Reflect.apply(target, thisArg, argArray)
+        }
+      }
+    })
+
+    const httpHook = new Hook(['http'], function onRequire (exports, name, basedir) {
+      exports.foo = 1
+      return exports
+    })
+    assert.equal(require('http').foo, 1, 'normal hooking still works')
+
+    const fnCore = require('@azure/functions-core')
+    assert.ok(fnCore, 'requiring monkey patched in module works')
+    assert.equal(fnCore.version, '1.0.0')
+    assert.equal(typeof fnCore.registerHook, 'function')
+
+    assert.throws(
+      () => require('package-does-not-exist'),
+      'Cannot find module \'package-does-not-exist\'',
+      'a failing `require(...)` can still throw as expected'
+    )
+
+    httpHook.unhook()
```

Comment:
This should go in an `afterEach` otherwise it won't be executed if the test fails.

Reply:
I added set up and tear down functions to the tests to avoid this issue